pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -1476,6 +1476,96 @@ func TestReconcileTimeouts(t *testing.T) {
 	}
 }
 
+func TestSortTaskRunStepOrder(t *testing.T) {
+	task := tb.Task("failing-task", "default", tb.TaskSpec(
+		tb.Step("hello", "busybox",
+			tb.Command("/bin/sh"), tb.Args("-c", "echo hello"),
+		),
+		tb.Step("exit", "busybox",
+			tb.Command("/bin/sh"), tb.Args("-c", "exit 1"),
+		),
+		tb.Step("world", "busybox",
+			tb.Command("/bin/sh"), tb.Args("-c", "sleep 30s"),
+		),
+	))
+
+	taskRunStatusSteps := []v1alpha1.StepState{{
+		ContainerState: corev1.ContainerState{
+			Terminated: &corev1.ContainerStateTerminated{
+				ExitCode: 0,
+				Reason:   "Completed",
+			},
+		},
+		Name: "world",
+
+	}, {
+		ContainerState: corev1.ContainerState{
+			Terminated: &corev1.ContainerStateTerminated{
+				ExitCode: 1,
+				Reason:   "Error",
+			},
+		},
+		Name: "exit",
+	}, {
+		ContainerState: corev1.ContainerState{
+			Terminated: &corev1.ContainerStateTerminated{
+				ExitCode: 0,
+				Reason:   "Completed",
+			},
+		},
+		Name: "hello",
+
+	},{
+
+		ContainerState: corev1.ContainerState{
+			Terminated: &corev1.ContainerStateTerminated{
+				ExitCode: 0,
+				Reason:   "Completed",
+			},
+		},
+		Name: "nop",
+	}}
+
+	expectedStepState := []v1alpha1.StepState{{
+		ContainerState: corev1.ContainerState{
+			Terminated: &corev1.ContainerStateTerminated{
+				ExitCode: 0,
+				Reason:   "Completed",
+			},
+		},
+		Name: "hello",
+	}, {
+		ContainerState: corev1.ContainerState{
+			Terminated: &corev1.ContainerStateTerminated{
+				ExitCode: 1,
+				Reason:   "Error",
+			},
+		},
+		Name: "exit",
+	}, {
+		ContainerState: corev1.ContainerState{
+			Terminated: &corev1.ContainerStateTerminated{
+				ExitCode: 0,
+				Reason:   "Completed",
+			},
+		},
+		Name: "world",
+	},{
+		ContainerState: corev1.ContainerState{
+			Terminated: &corev1.ContainerStateTerminated{
+				ExitCode: 0,
+				Reason:   "Completed",
+			},
+		},
+		Name: "nop",
+	}}
+
+	sortTaskRunStepOrder(taskRunStatusSteps, task.Spec.Steps)
+	if d := cmp.Diff(taskRunStatusSteps, expectedStepState); d != "" {
```

Comment:
this is pretty minor but the test would  be a bit easier to read if it asserted the names of the names of the steps only, e.g. something like:

```
actualStepOrder := []string{}
for _, state :=  range(taskRunStatusSteps) {
    actualStepOrder = append(actualStepOrder, state.name)
}

expectedStepOrder := []string{"hello", "exit", "world", "nop"}
if d := cmp.Diff(actualStepOrder, expectedStepOrder); d != "" {
}
```

Then you can completely get rid of `expectedStepState` :D

Reply:
(this would also make it a bit easier to add a few more tests cases as well)

Reply:
Done. Thx.